solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -1831,21 +1814,27 @@ private void fetch() throws Exception {
     private int fetchPackets(FastInputStream fis) throws Exception {
       byte[] intbytes = new byte[4];
       byte[] longbytes = new byte[8];
+      boolean isContentReceived = false;
       try {
         while (true) {
+          if (fis.peek() == -1) {
+            return NO_CONTENT;
+          }
           if (stop) {
             stop = false;
             aborted = true;
             throw new ReplicationHandlerException("User aborted replication");
           }
           long checkSumServer = -1;
+
           fis.readFully(intbytes);
           // read the size of the packet
           int packetSize = readInt(intbytes);
           if (packetSize <= 0) {
-            log.warn("No content received for file: {}", fileName);
-            return NO_CONTENT;
+            if (!isContentReceived) log.warn("No content received for file: {}", fileName);
```

Comment:
Since we don't return here, this probably isn't where we should log this message.
Nonetheless if we get to this spot, it's now highly unexpected since even if there was no content, the peek==-1 above would happen and we wouldn't get here. So what do we do?  Hmm.  If we think it's impossible (I think?), add an assert.  Could also log a warning.